### PR TITLE
Improved metadata handling in readSource + metadata helper function

### PR DESCRIPTION
**Pull request by tscheypidi**
- adapted metadata handling from calcOutput in readSource (metadata will now be added as comment to the returned data objects)
- added helper function getMetadataComment to more easily extract metadata from a data object

### CODE REVIEW

File: R/prepExtendedComment.R [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#'
+#' Helper function condense metadata information into
+#' an extended comment entry
+#'
+#' @md
+#' @param x list containing the metadata to be condensed
+#' @param type output type, e.g. "TauTotal"
+#' @param warn boolean indicating whether warnings should be triggered
+#' if entries are missing, or not.
+#' @param n the number of generations to go back
```
Comment (pascal-sauer) [inline R/prepExtendedComment.R:11]: What is a generation here? I feel this arg's documentation needs more clarification

File: R/prepExtendedComment.R [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  cl <- sys.call(-n)
+  f <- get(as.character(cl[[1]]), mode = "function", sys.frame(-n - 1))
+  cl <- match.call(definition = f, call = cl)
+
+  if (isTRUE(warn)) {
+    unitWarning <- paste0('Missing unit information for data set "', type, '"!')
+    descriptionWarning <- paste0('Missing description for data set "', type,
+                                 '"! Please add a description in the corresponding calc function!')
+  } else {
+    unitWarning <- descriptionWarning <- NULL
```
Comment (pascal-sauer) [inline R/prepExtendedComment.R:30]:
```suggestion
    unitWarning <- NULL
    descriptionWarning <- NULL
```

File: R/prepExtendedComment.R [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#' test <- function(a = 1) {
+#'    return(madrat:::prepExtendedComment(list(unit = "m", description = "example", package = "blub")))
+#'  }
+#'  test(a = 42)
+#'
+prepExtendedComment <- function(x, type = "#undefined", warn = TRUE, n = 1) {
+
+  cl <- sys.call(-n)
+  f <- get(as.character(cl[[1]]), mode = "function", sys.frame(-n - 1))
+  cl <- match.call(definition = f, call = cl)
```
Comment (pascal-sauer) [inline R/prepExtendedComment.R:23]: Maybe a comment to clarify what is happening here?

File: R/cleanComment.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#' cleanComment
+#'
+#' Helper function to clean a comment from additional metadata information
+#'
+#' @md
+#' @param x magclass object the comment shold be read from
```
Comment (pascal-sauer) [inline R/cleanComment.R:6]:
```suggestion
#' @param x magclass object the comment should be read from
```

File: R/cleanComment.R [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#' @param remove Vector of categories to be removed
+#' @author Jan Philipp Dietrich
+#' @examples
+#' x <- maxample("animal")
+#' getComment(x) <- c("unit: bla","comment: hallo","blub: ble")
```
Comment (pascal-sauer) [inline R/cleanComment.R:11]:
```suggestion
#' getComment(x) <- c("unit: bla", "comment: hallo", "blub: ble")
```

File: R/getMetadataComment.R [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#' getMetadataComment
+#'
+#' Helper function extract a metadata comment
```
Comment (pascal-sauer) [inline R/getMetadataComment.R:3]:
```suggestion
#' Helper function to extract metadata from a magclass object's comment.
```

File: R/getMetadataComment.R [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+#' @param name name of the metadata to be extracted (e.g. unit)
+#' @author Jan Philipp Dietrich
+#' @examples
+#' x <- as.magpie(1)
+#' getComment(x) <- c(" description: example description", " unit: kg")
+#' getMetadataComment(x, "unit")
+#' getMetadataComment(x, "description")
+#' @export
+
+getMetadataComment <- function(x, name) {
```
Comment (pascal-sauer) [inline R/getMetadataComment.R:16]:
```suggestion
getMetadata <- function(x, name) {
```
I find "Comment" a bit misleading, it reads to me like you can extract the "comment" from the metadata. It is actually `getMetadataFromComment`, right? Maybe to keep it short just `getMetadata`?

File: R/getMetadataComment.R [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  comment <- attr(x, "comment")
+  key <- paste0("^ ", name, ": ")
+  entry <- grep(key, comment)
+  if (length(entry) == 0) return(NULL)
```
Comment (pascal-sauer) [inline R/getMetadataComment.R:20]: Did you consider throwing an error or warning in this case? Maybe one that can be disabled with an argument

File: R/getMetadataComment.R (hunk 6 above)
Comment (tscheypidi) [reply R/getMetadataComment.R:3]: actually it also works for other objects as we create a comment attribute also for non-magclass objects

File: R/getMetadataComment.R (hunk 8 above)
Comment (tscheypidi) [reply R/getMetadataComment.R:20]: thought about it, but decided not to implement it at this stage. We can add this feature later in case we need it

**Review by pascal-sauer — APPROVED** [review]

**Pull request merged**